### PR TITLE
build: Wire hipFile unit tests into TheRock test flow

### PR DESCRIPTION
**Pull request by derobins**
## Motivation

hipFile has been added to TheRock, but still lacks proper CI and test integration.

## Technical Details

Run hipFile's unit tests from packaged artifacts as part of TheRock CI. hipFile (in rocm-systems) installs its unit-test suite as a relocatable tree under share/hipfile/test; this change consumes that on the TheRock side so the tests are packaged, fetched, and run like any other component.

- artifact-hipfile.toml: capture the relocatable ctest tree into the `test` component (share/hipfile/test/**); remove the now-dead share/hipfile/** glob from `lib`.
- storage-libs/CMakeLists.txt: pass -DAIS_INSTALL_TESTS=${THEROCK_BUILD_TESTING} so the hipFile subproject installs its tests in TheRock builds.
- install_rocm_from_artifacts.py: add a --hipfile flag to fetch the hipFile lib (and test artifacts under --tests).
- test_hipfile.py: new runner that invokes `ctest --test-dir <install>/share/hipfile/test -L unit`; the installed tree resolves its own library path, so no environment setup is needed.
- fetch_test_configurations.py: add a `hipfile` test-matrix entry (--hipfile --tests, linux-only, single shard, 15-min timeout) since the unit tests are CPU-only and need no GPU runner.

Depends on the rocm-systems hipFile test-install change; the `test` component is empty until that lands and the submodule is bumped.  This will be a draft until https://github.com/ROCm/rocm-systems/pull/7590 is merged and visible in a submodule bump. 

JIRA ID: AIHIPFILE-27
JIRA ID: ROCM-1261

## Test Plan

CI passes, manual inspection of artifacts

## Test Result

Success

## Submission Checklist

- [x] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

### CODE REVIEW

**Comment by derobins** [comment]
This will be a draft until https://github.com/ROCm/rocm-systems/pull/7590 is merged and visible in a submodule bump. The "draft" status is just to keep it from being merged, but this is otherwise complete and can be reviewed.

File: build_tools/github_actions/fetch_test_configurations.py [hunk 1]
```diff
@@ -156,6 +156,18 @@ def _build_container_options(job_config: dict, platform: str) -> dict:
...
+    "hipfile": {
+        "job_name": "hipfile",
+        "fetch_artifact_args": "--hipfile --tests",
+        "timeout_minutes": 15,
+        "test_script": f"python {_get_script_path('test_hipfile.py')}",
+        "platform": ["linux"],
+        "total_shards_dict": {
+            "linux": 1,
+        },
+    },
```
[the remark this answers is not in the document]
Comment (ammallya) [reply build_tools/github_actions/fetch_test_configurations.py:171]: Please apply this

File: build_tools/github_actions/test_executable_scripts/test_hipfile.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    logging.error(f"hipFile test directory not found: {HIPFILE_TEST_DIR}")
+    raise SystemExit(1)
+
+cmd = [
+    "ctest",
+    "--test-dir",
+    str(HIPFILE_TEST_DIR),
+    "-L",
+    "unit",
+    "--output-on-failure",
```
Comment (ammallya) [inline build_tools/github_actions/test_executable_scripts/test_hipfile.py:45]: Please add "--no-tests=error", to catch empty test set

File: build_tools/github_actions/test_executable_scripts/test_hipfile.py
Comment (geomin12) [inline build_tools/github_actions/test_executable_scripts/test_hipfile.py:1]: test fails https://github.com/ROCm/TheRock/actions/runs/27962494637/job/82774104639?pr=6026#step:13:35 

:(

File: build_tools/github_actions/fetch_test_configurations.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (geomin12) [reply build_tools/github_actions/fetch_test_configurations.py:171]: i also agree

File: build_tools/github_actions/test_executable_scripts/test_hipfile.py
Comment (gaoikawa) [reply build_tools/github_actions/test_executable_scripts/test_hipfile.py:1]: @geomin12 Test here is expected to fail until the underlying dependent PR is merged and the bump PR picks it up.

File: build_tools/github_actions/test_executable_scripts/test_hipfile.py (hunk 2 above)
Comment (derobins) [reply build_tools/github_actions/test_executable_scripts/test_hipfile.py:45]: Done

File: build_tools/github_actions/fetch_test_configurations.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (derobins) [reply build_tools/github_actions/fetch_test_configurations.py:171]: Done

**Review by ammallya — APPROVED** [review]
LGTM

**Comment by ScottTodd** [comment]
Manually triggered test run to skip the CI rebuild time: https://github.com/ROCm/TheRock/actions/runs/28057929998/job/83065263123

```
520/521 Test #520: StreamSuite/HipFileStreamValidParams.register_stream_valid_flags_internal/14 .....................................   Passed    0.02 sec
        Start 521: StreamSuite/HipFileStreamValidParams.register_stream_valid_flags_internal/15
521/521 Test #521: StreamSuite/HipFileStreamValidParams.register_stream_valid_flags_internal/15 .....................................   Passed    0.02 sec
100% tests passed, 0 tests failed out of 521
Label Time Summary:
internal    =  15.82 sec*proc (521 tests)
unit        =  15.82 sec*proc (521 tests)
Total Test time (real) =  16.11 sec
```

**Review by marbre — APPROVED** [review]
With https://github.com/ROCm/TheRock/actions/runs/28057929998 this lgtm!
